lib/jobs/copy-publickey.js

```diff
@@ -0,0 +1,66 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+module.exports = copyKeyJobFactory;
+
+di.annotate(copyKeyJobFactory, new di.Provide('Job.CopyKey'));
+di.annotate(copyKeyJobFactory, new di.Inject(
+    'Job.Base',
+    'Util',
+    'Logger',
+    'Assert',
+    'Promise',
+    'Services.Waterline',
+    'ssh',
+    'JobUtils.Commands'
+));
+
+function copyKeyJobFactory(
+    BaseJob,
+    util,
+    Logger,
+    assert,
+    Promise,
+    waterline,
+    ssh,
+    CommandUtil
+) {
+    var logger = Logger.initialize(copyKeyJobFactory);
+    var commandUtil;
+    function CopyKeyJob(options, context, taskId) {
+        CopyKeyJob.super_.call(this, logger, options, context, taskId);
+        assert.string(this.context.target);
+        this.nodeId = this.context.target;
+        commandUtil = new CommandUtil(this.nodeId);
+    }
+    util.inherits(CopyKeyJob, BaseJob);
+
+    CopyKeyJob.prototype._run = function run() {
+        var self = this;
+        return waterline.nodes.needByIdentifier(self.nodeId)
+        .then(function(node) {
+            assert.string(node.sshSettings.publicKey);
+            var commands = [
+                {cmd: 'mkdir -p .ssh'},
+                {cmd: 'echo '+node.sshSettings.publicKey+' >> .ssh/authorized_keys'}
```

Comment:
Yeah, ubuntu default is:
`-rw------- 1 vagrant vagrant 389 Apr 11 20:45 /home/vagrant/.ssh/authorized_keys`